core/include/thread.h

```diff
@@ -11,6 +11,100 @@
  * @ingroup     core
  * @brief       Support for multi-threading
  *
+ * Priorities
+ * ==========
+ *
+ * As RIOT is using a fixed priority @ref core_sched "scheduling algorithm",
+ * threads are scheduled based on their priority. The priority is fixed for
+ * every thread and specified during the threads creation by the `priority`
+ * parameter.
+ *
+ * The lower the priority number, the higher the priority of the thread,
+ * with 0 being the highest possible priority.
+ *
+ * The lowest possible priority is @ref THREAD_PRIORITY_IDLE - 1. The exact value
+ * depends on the platforms architecture, e.g. it would be 30 in 32-bit systems,
+ * and 14 in 16-bit systems.
+ *
+ * @note Assigning the same priority to two or more threads is usually not a
+ *       good idea. A thread in RIOT may run until it yields (@ref
+ *       thread_yield) or another thread with higher priority is runnable (@ref
+ *       STATUS_ON_RUNQUEUE) again. Having multiple threads with the same
+ *       priority may make it difficult to determine when which of them gets
+ *       scheduled and how much CPU time they will get. In most applications,
+ *       the number of threads in application is significantly smaller than the
+ *       number of available priorities, so assigning distinct priorities per
+ *       thread should not be a problem. Only assign the same priority to
+ *       multiple threads if you know what you are doing!
+ *
+ * Thread Behavior
+ * ===============
+ * In addition to the priority, flags can be used when creating a thread to
+ * alter the thread's behavior after creation. The following flags are available:
+ *
+ *  @ref THREAD_CREATE_SLEEPING :   the newly created thread will be put to sleeping
```

Comment:
Try

```diff
diff --git a/core/include/thread.h b/core/include/thread.h
index 7ab4fcf..83d72c6 100644
--- a/core/include/thread.h
+++ b/core/include/thread.h
@@ -42,15 +42,11 @@
  * In addition to the priority, flags can be used when creating a thread to
  * alter the thread's behavior after creation. The following flags are available:
  *
- *  @ref THREAD_CREATE_SLEEPING :   the newly created thread will be put to sleeping
- *                                  state and must be woken up manually
- *
- *  @ref THREAD_CREATE_WOUT_YIELD : the newly created thread will not run
- *                                  immediately after creation
- *
- *  @ref THREAD_CREATE_STACKTEST :  write markers into the thread's stack to measure
- *                                  the stack's memory usage (for debugging and
- *                                  profiling purposes)
+ *  Flags                         | Description
+ *  ----------------------------- | ----------------------------------------------------------------
+ *  @ref THREAD_CREATE_SLEEPING   | the newly created thread will be put to sleeping state
+ *  @ref THREAD_CREATE_WOUT_YIELD | the newly created thread will not run immediately after creation
+ *  @ref THREAD_CREATE_STACKTEST  | measures the stack's memory usage
  *
  * Thread creation
  * ===============
```

It's not perfect because we loose some information (there is no multi-lining possible in those tables), but I think it looks a little better in HTML and there are still the refs to get more detailed information ;-).